src/platform/with_eagl/native_gl_context.rs

```diff
@@ -0,0 +1,142 @@
+use platform::NativeGLContextMethods;
+use GLVersion;
+use gleam::gl;
+use objc::runtime::{BOOL, NO};
+use objc::runtime::{Class, Object};
+use libloading as lib;
+use std::ops::Deref;
+use std::ptr;
+
+type EAGLContext = *mut Object;
+type EAGLSharegroup = *mut Object;
+
+lazy_static! {
+    static ref EAGLCONTEXT_CLASS: &'static Class  = {
+        Class::get("EAGLContext").expect("EAGLContext class not found")
+    };
+
+    static ref OPENGLES_FRAMEWORK: Option<lib::Library>  = {
+        lib::Library::new("/System/Library/Frameworks/OpenGLES.framework/OpenGLES").ok()
+    };
+}
+
+pub struct NativeGLContext(EAGLContext);
+pub type NativeGLContextHandle = NativeGLContext;
+unsafe impl Send for NativeGLContext {}
+
+impl NativeGLContext {
+    fn new_retained(context: EAGLContext) -> Self {
+        let context = unsafe { msg_send![context, retain] };
+        NativeGLContext(context)
+    }
+}
+
+impl Drop for NativeGLContext {
+    fn drop(&mut self) {
+        unsafe {
+            msg_send![self.0, release];
+        }
+    }
+}
+
+impl NativeGLContextMethods for NativeGLContext {
+    type Handle = Self;
+
+    fn get_proc_address(addr: &str) -> *const () {
+        let framework = match *OPENGLES_FRAMEWORK {
+            Some(ref lib) => lib,
+            None => return ptr::null(),
+        };
+        unsafe {
+            let symbol: Result<lib::Symbol<unsafe extern fn()>, _> = framework.get(addr.as_bytes());
+            match symbol {
+                Ok(symbol) => *symbol.deref() as *const (),
+                _ => ptr::null()
+            }
+        }
+    }
+
+    fn create_shared(with: Option<&Self::Handle>,
+                     api_type: &gl::GlType,
+                     api_version: GLVersion) -> Result<NativeGLContext, &'static str> {
+        match *api_type {
```

Comment:
Should this be the other way around? i.e. `if api_type != GlType::Gles` instead of the opposite?

Reply:
(I know it's equivalent, just maybe a bit simpler)